### PR TITLE
Add NNotepad sample

### PR DESCRIPTION
**Pull request by inexorabletash**
It is not yet linked from the main README or index page as there is significant integration/cleanup work to do. But it is useful as-is for WebNN implementers.

A bespoke test harness is included. Documentation visible in the sample is generated from the README.md (run `make` to update).

### CODE REVIEW

**Comment by inexorabletash** [comment]
@huningxin can you take a look, or recommend other reviewers?

**Review by huningxin — APPROVED** [review]
LGTM! Thanks @inexorabletash , I like your demo so much, I can't wait for trying it.

/cc @ibelem @Honry

File: nnotepad/TODO.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# To-Do
+
+## Basics
+
+* Style to match rest of webnn-samples.
+* Improve default text.
+* Consider incorporating [WebNN Polyfill](https://github.com/webmachinelearning/webnn-polyfill).
+* Make input/output areas resizable.
+* Add to `../README.md` once we're happy with it.
+
```
Comment (Honry) [inline nnotepad/TODO.md:10]: - Style to match rest of webnn-samples.
 - Consider incorporating [WebNN Polyfill](https://github.com/webmachinelearning/webnn-polyfill).

I can take these two TODOs.

**Comment by huningxin** [comment]
https://webmachinelearning.github.io/webnn-samples/nnotepad/ is live!

/cc @anssiko

**Pull request merged**